src/gc/proxy.d

```diff
@@ -13,106 +13,79 @@
  */
 module gc.proxy;
 
-import gc.gc;
+import conservative = gc.impl.conservative.gc;
+import manual = gc.impl.manual.gc;
+import gc.config;
 import gc.stats;
-import core.stdc.stdlib;
 
 private
 {
-    __gshared GC _gc;
-
     static import core.memory;
     alias BlkInfo = core.memory.GC.BlkInfo;
 
     extern (C) void thread_init();
     extern (C) void thread_term();
 
-    struct Proxy
-    {
-        extern (C)
-        {
-            void function() gc_enable;
-            void function() gc_disable;
-        nothrow:
-            void function() gc_collect;
-            void function() gc_minimize;
-
-            uint function(void*) gc_getAttr;
-            uint function(void*, uint) gc_setAttr;
-            uint function(void*, uint) gc_clrAttr;
-
-            void*   function(size_t, uint, const TypeInfo) gc_malloc;
-            BlkInfo function(size_t, uint, const TypeInfo) gc_qalloc;
-            void*   function(size_t, uint, const TypeInfo) gc_calloc;
-            void*   function(void*, size_t, uint ba, const TypeInfo) gc_realloc;
-            size_t  function(void*, size_t, size_t, const TypeInfo) gc_extend;
-            size_t  function(size_t) gc_reserve;
-            void    function(void*) gc_free;
-
-            void*   function(void*) gc_addrOf;
-            size_t  function(void*) gc_sizeOf;
-
-            BlkInfo function(void*) gc_query;
-
-            void function(void*) gc_addRoot;
-            void function(void*, size_t, const TypeInfo ti) gc_addRange;
+    __gshared Proxy*  pthis;
```

Comment:
What I meant was, if you set a new proxy and then call gc_clrProxy, pthis is what is fallen back on after the current proxy is cleared.